src/basic/stat-util.c

```diff
@@ -205,7 +205,9 @@ int path_check_fstype(const char *path, statfs_f_type_t magic_value) {
 
 bool is_temporary_fs(const struct statfs *s) {
     return is_fs_type(s, TMPFS_MAGIC) ||
-           is_fs_type(s, RAMFS_MAGIC);
+           is_fs_type(s, RAMFS_MAGIC) ||
+           is_fs_type(s, CGROUP_SUPER_MAGIC) ||
+           is_fs_type(s, CGROUP2_SUPER_MAGIC);
```

Comment:
i figure it would make sense to add a cg_is_cgroupfs() call to cgroup-util.h, and then call that from rm-rf.c and also allow it.